src/core/Slider/Slider.tsx

```diff
@@ -439,9 +439,13 @@ export const Slider = React.forwardRef<HTMLDivElement, SliderProps>(
           <div className='iui-slider-rail' />
           {currentValues.map((thumbValue, index) => {
             const [minVal, maxVal] = getAllowableThumbRange(index);
+            const thisThumbProps = thumbProps?.(index) ?? {};
+            const key = thisThumbProps.id
+              ? thisThumbProps.id
+              : index.toString();
             return (
               <Thumb
-                key={index}
+                key={key}
```

Comment:
Does this not work?

```suggestion
                key={`${thumbProps?.(index)?.id ?? index}`}
```

Reply:
I like being able to see the value in debugger... but I updated to mostly use your suggestion. I see no reason to run `thumbProps?.(index)` twice so I still create `thisThumbProps` and use if for the `key` and for `thumbProps` .

Reply:
Well, if you're assigning `{}` anyway, then you don't need `?.`

So you can change it to `key={thisThumbProps.id ?? index}`